[BUG] PydanticDeprecatedSince20 warning for 'min_items' in mlflow.types.chat.BaseRequestPayload with Pydantic V2

### Issues Policy acknowledgement

- [x] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Where did you encounter this bug?

Local machine

### MLflow version

MLflow Version: 2.22.0


### System information


pydantic: 2.11.4         
pydantic-core: 2.33.2 
Python Version: 3.11.12
Operating System: Fedora Workstation
MLflow installed in a Conda environment


### Describe the problem

When using mlflow==2.22.0 with a Pydantic V2.x environment, PydanticDeprecatedSince20 warnings are triggered due to mlflow's internal use of min_items in its Pydantic models.
Mention that Pydantic V2 recommends using min_length instead of min_items.

### Tracking information

<!-- PLEASE KEEP BACKTICKS AND CHECK PREVIEW -->
```shell
REPLACE_ME
```


### Code to reproduce issue

<!-- PLEASE KEEP BACKTICKS AND CHECK PREVIEW -->
```
# Filename: reproduce_mlflow_pydantic_warning_v2.py

import warnings
import importlib
import sys

# Attempt to import the specific Pydantic warning class
PydanticDeprecatedSince20 = None
PYDANTIC_WARNING_CLASS_AVAILABLE = False
pydantic_version = "Not available"

try:
    # Import the pydantic.warnings submodule
    pydantic_warnings_module = importlib.import_module('pydantic.warnings')
    # Access the PydanticDeprecatedSince20 class from this submodule
    PydanticDeprecatedSince20 = pydantic_warnings_module.PydanticDeprecatedSince20
    PYDANTIC_WARNING_CLASS_AVAILABLE = True

    # Get Pydantic version for context
    pydantic_core_module = importlib.import_module('pydantic')
    pydantic_version = pydantic_core_module.__version__

except ImportError:
    print("Failed to import 'pydantic.warnings' or 'PydanticDeprecatedSince20' class not found in it.")
    print("Please ensure Pydantic v2+ is installed correctly.")
except AttributeError:
    print("Failed to find 'PydanticDeprecatedSince20' within 'pydantic.warnings'.")
    print("This might indicate an unexpected Pydantic version or structure.")

# Try to import mlflow to get its version and then its types
mlflow_version = "Not available"
try:
    mlflow_module = importlib.import_module('mlflow')
    MLFLOW_AVAILABLE = True
    mlflow_version = mlflow_module.__version__
except ImportError:
    print("MLflow library is not available. This script requires MLflow to run.")
    MLFLOW_AVAILABLE = False

if not PYDANTIC_WARNING_CLASS_AVAILABLE or not MLFLOW_AVAILABLE:
    print("Exiting due to missing prerequisites (Pydantic warning class or MLflow).")
    sys.exit(1)

print(f"--- System Information ---")
print(f"Python version: {sys.version.split()[0]}")
print(f"MLflow version: {mlflow_version}")
print(f"Pydantic version: {pydantic_version}")
print(f"--------------------------\n")

warnings.filterwarnings(
    "error",
    message=r".*`min_items` is deprecated.*use `min_length` instead.*", # Updated regex
    category=PydanticDeprecatedSince20 # Use the directly imported class
)

print("Attempting to import 'mlflow.types.chat'...")
print("This import is expected to trigger the PydanticDeprecatedSince20 warning for 'min_items' from within MLflow's codebase.")

try:
    import mlflow.types.chat

    print("\nImport of 'mlflow.types.chat' was successful.")
    print("If the PydanticDeprecatedSince20 warning was not raised as an error,")
    print("it might indicate that the warning is filtered differently in this script,")
    print("the issue is not triggered by this specific import path alone in this context,")
    print("or it has been fixed in your current library versions.")

except PydanticDeprecatedSince20 as e: # Use the directly imported class here too
    print(f"\n>>> SUCCESS: Caught the expected PydanticDeprecatedSince20 warning as an error <<<")
    print(f"Error message: {e}")
    print("\nThis confirms that importing 'mlflow.types.chat' (or its dependencies)")
    print("triggers the deprecated use of 'min_items' within the MLflow library.")
    print("The traceback for this error (if run directly) will point to the MLflow source code.")

except Exception as e_other:
    print(f"\nAn unexpected error occurred during the import of 'mlflow.types.chat':")
    print(f"{type(e_other).__name__}: {e_other}")
    print("This could be due to other issues or if the warning was not correctly filtered as an error.")

```


### Stack trace

<!-- PLEASE KEEP BACKTICKS AND CHECK PREVIEW -->
```
================================================================================ test session starts =================================================================================
platform linux -- Python 3.11.12, pytest-8.3.5, pluggy-1.5.0
rootdir: /home/eric/Bureau/project_p7
plugins: anyio-4.9.0
collected 0 items / 1 error                                                                                                                                                          

======================================================================================= ERRORS =======================================================================================
_________________________________________________________________________ ERROR collecting tests/test_app.py _________________________________________________________________________
tests/test_app.py:16: in <module>
    from credit_risk_app.main import app
src/credit_risk_app/main.py:15: in <module>
    import mlflow.pyfunc
../../.anaconda3/envs/credit_risk_env/lib/python3.11/site-packages/mlflow/pyfunc/__init__.py:427: in <module>
    import mlflow.pyfunc.loaders
../../.anaconda3/envs/credit_risk_env/lib/python3.11/site-packages/mlflow/pyfunc/loaders/__init__.py:1: in <module>
    import mlflow.pyfunc.loaders.chat_agent
../../.anaconda3/envs/credit_risk_env/lib/python3.11/site-packages/mlflow/pyfunc/loaders/chat_agent.py:8: in <module>
    from mlflow.pyfunc.model import (
../../.anaconda3/envs/credit_risk_env/lib/python3.11/site-packages/mlflow/pyfunc/model.py:40: in <module>
    from mlflow.types.agent import (
../../.anaconda3/envs/credit_risk_env/lib/python3.11/site-packages/mlflow/types/agent.py:5: in <module>
    from mlflow.types.chat import BaseModel, ChatUsage, ToolCall
../../.anaconda3/envs/credit_risk_env/lib/python3.11/site-packages/mlflow/types/chat.py:157: in <module>
    class BaseRequestPayload(BaseModel):
../../.anaconda3/envs/credit_risk_env/lib/python3.11/site-packages/mlflow/types/chat.py:162: in BaseRequestPayload
    stop: Optional[list[str]] = Field(None, min_items=1)
../../.anaconda3/envs/credit_risk_env/lib/python3.11/site-packages/pydantic/fields.py:1045: in Field
    warn('`min_items` is deprecated and will be removed, use `min_length` instead', DeprecationWarning)
E   pydantic.warnings.PydanticDeprecatedSince20: `min_items` is deprecated and will be removed, use `min_length` instead. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.11/migration/
============================================================================== short test summary info ===============================================================================
ERROR tests/test_app.py - pydantic.warnings.PydanticDeprecatedSince20: `min_items` is deprecated and will be removed, use `min_length` instead. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pyda...
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
================================================================================== 1 error in 1.35s ==================================================================================


```


### Other info / logs

<!-- PLEASE KEEP BACKTICKS AND CHECK PREVIEW -->
```
REPLACE_ME
```


### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [x] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

--- a/mlflow/gateway/providers/cohere.py
+++ b/mlflow/gateway/providers/cohere.py
@@ -191,7 +191,7 @@ class CohereAdapter(ProviderAdapter):
         payload = cls._scale_temperature(payload)
 
         messages = payload.pop("messages")
-        last_message = messages.pop()  # pydantic enforces min_items=1
+        last_message = messages.pop()  # pydantic enforces min_length=1
         if last_message["role"] != "user":
             raise AIGatewayException(
                 status_code=422,

--- a/mlflow/gateway/schemas/chat.py
+++ b/mlflow/gateway/schemas/chat.py
@@ -66,7 +66,9 @@ _REQUEST_PAYLOAD_EXTRA_SCHEMA = {
 
 
 class RequestPayload(ChatCompletionRequest, RequestModel):
-    messages: list[RequestMessage] = Field(..., min_length=1)
+    messages: list[RequestMessage] = (
+        Field(..., min_length=1) if IS_PYDANTIC_V2_OR_NEWER else Field(..., min_items=1)
+    )
     tools: Optional[list[ChatToolWithUC]] = None
 
     class Config:

--- a/mlflow/gateway/schemas/chat.py
+++ b/mlflow/gateway/schemas/chat.py
@@ -66,7 +66,7 @@ _REQUEST_PAYLOAD_EXTRA_SCHEMA = {
 
 
 class RequestPayload(ChatCompletionRequest, RequestModel):
-    messages: list[RequestMessage] = Field(..., min_items=1)
+    messages: list[RequestMessage] = Field(..., min_length=1)
     tools: Optional[list[ChatToolWithUC]] = None
 
     class Config:

--- a/mlflow/types/chat.py
+++ b/mlflow/types/chat.py
@@ -159,7 +159,7 @@ class BaseRequestPayload(BaseModel):
 
     temperature: float = Field(0.0, ge=0, le=2)
     n: int = Field(1, ge=1)
-    stop: Optional[list[str]] = Field(None, min_items=1)
+    stop: Optional[list[str]] = Field(None, min_length=1)
     max_tokens: Optional[int] = Field(None, ge=1)
     stream: Optional[bool] = None
     model: Optional[str] = None
@@ -213,8 +213,8 @@ class ChatCompletionRequest(BaseRequestPayload):
     https://platform.openai.com/docs/api-reference/chat
     """
 
-    messages: list[ChatMessage] = Field(..., min_items=1)
-    tools: Optional[list[ChatTool]] = Field(None, min_items=1)
+    messages: list[ChatMessage] = Field(..., min_length=1)
+    tools: Optional[list[ChatTool]] = Field(None, min_length=1)
 
 
 class ChatCompletionResponse(BaseModel):

--- a/mlflow/types/chat.py
+++ b/mlflow/types/chat.py
@@ -159,7 +159,9 @@ class BaseRequestPayload(BaseModel):
 
     temperature: float = Field(0.0, ge=0, le=2)
     n: int = Field(1, ge=1)
-    stop: Optional[list[str]] = Field(None, min_length=1)
+    stop: Optional[list[str]] = (
+        Field(None, min_length=1) if IS_PYDANTIC_V2_OR_NEWER else Field(None, min_items=1)
+    )
     max_tokens: Optional[int] = Field(None, ge=1)
     stream: Optional[bool] = None
     model: Optional[str] = None
@@ -213,8 +215,12 @@ class ChatCompletionRequest(BaseRequestPayload):
     https://platform.openai.com/docs/api-reference/chat
     """
 
-    messages: list[ChatMessage] = Field(..., min_length=1)
-    tools: Optional[list[ChatTool]] = Field(None, min_length=1)
+    messages: list[ChatMessage] = (
+        Field(..., min_length=1) if IS_PYDANTIC_V2_OR_NEWER else Field(..., min_items=1)
+    )
+    tools: Optional[list[ChatTool]] = (
+        Field(None, min_length=1) if IS_PYDANTIC_V2_OR_NEWER else Field(None, min_items=1)
+    )
 
 
 class ChatCompletionResponse(BaseModel):

